### ISSUE TITLE
DataTable: lazyLoading fetches rows two times if updated

### ISSUE THREAD

**Issue by malex666** [primefaces/primefaces#1009]
We have been struggling with a very unusual problen for days now. For the setup: create a DataTable with eg. 400 entries, lazy,scrollable and liveScroll = true and scroll rows = 25. Create a <p:commandButton> which updates the datatable by id.

index.xhtml

```
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml"
    xmlns:ui="http://java.sun.com/jsf/facelets"
    xmlns:h="http://java.sun.com/jsf/html"
    xmlns:f="http://java.sun.com/jsf/core"
    xmlns:p="http://primefaces.org/ui">

<f:view contentType="text/html">
    <h:head>
    </h:head>
    <h:body>
        <h:form id="form">
            <p:commandButton value="breakStuff" update=":form:table" process="@this" partialSubmit="true"/>
            <p:dataTable id="table" value="#{bean.cars}" var="car" style="width:200px" 
                scrollable="true" scrollHeight="400"  lazy="true"
                scrollRows="25">
                <p:column headerText="#1">
                    <h:outputText value="#{car.name}" />
                </p:column>
                <p:column headerText="#2">
                    <h:outputText value="#{car.description}" />
                </p:column>
            </p:dataTable>
        </h:form>
    </h:body>
</f:view>
</html>
```

CarUiBean.java

```
package test;

import java.io.Serializable;
import java.util.List;

import javax.annotation.PostConstruct;
import javax.faces.view.ViewScoped;
import javax.inject.Named;

@Named("bean")
@ViewScoped
public class CarUiBean implements Serializable {
    private static final long serialVersionUID = 3684508986415928124L;
    List<Car> cars;

    @PostConstruct
    public void init() {
        cars = Car.createDummyCars(2000);
    }

    public List<Car> getCars() {
        return cars;
    }
}
```

Car.java

```
package test;

import java.util.ArrayList;
import java.util.List;

public class Car {
    private String name;
    private String description;

    private Car(String name, String description) {
        this.name = name;
        this.description = description;
    }

    public static List<Car> createDummyCars(int amnt){
        List<Car> cars = new ArrayList<>();
        for(int i = 0; i < amnt; i++){
            cars.add(new Car("TEST", String.valueOf(i)));
        }
        return cars;
    }

    public String getName() {
        return this.name;
    }

    public String getDescription() {
        return this.description;
    }
}
```

Now scroll to the bottom of the dataTable (until it has lazyLoaded all entries), click the button which just updates the datatable and then scroll down again, the entries will start from the first again at the very end. Eg. if you have like 400 entries and you scroll down afterwards it starts again from 1 after 400. Like 399, 400, 1, 2,3, 4... . It seems like the data is doubled in the dataTable after updating?!

**Comment by kukel** [comment]
What PF version?

**Comment by malex666** [comment]
Sorry, forgot about that. We tried with 5.3 community, 5.3.3 elite and 5.3.4 elite. Didn't change anything.

**Comment by kukel** [comment]
Please post al this in the forum. It does not belong in the issues in github. You use 'lazy' but do not implement the lazyDataModel Please look at the showcase and the docs!!! Please delete this 'issue'

**Comment by malex666** [comment]
Hi,
it doesn't matter if I use lazy true or false, even if I just leave the lazy option the same problem occurs, so this can't be a problem related to the lazyDataModel or the lazy option. Seems like this 'issue' is related to the liveScroll option

```
<!-- without lazy option, problem still occurs -->
<p:dataTable id="table" value="#{bean.cars}" var="car"
                style="width:200px" scrollable="true" scrollHeight="400"
                liveScroll="true" scrollRows="25">
                <p:column headerText="#1">
                    <h:outputText value="#{car.name}" />
                </p:column>
                <p:column headerText="#2">
                    <h:outputText value="#{car.description}" />
                </p:column>
            </p:dataTable>
```

**Comment by malex666** [comment]
any news? Datatables are highly unusable if you can't update the table as it is now.

**Comment by Maikel-Nait** [comment]
FYI in our application we had some other issues using liveScroll with datatables, and finally replaced the [p:dataTable] with a [p:dataScroller] component, (it was not that hard to replace ) which seems to work just fine.  Maybe it can be a valid workaround for you while this problem is fixed ...

**Comment by noxid666** [comment]
thanks for your answer. Good to know that it's not just us who do have problems with live scroll. Your approach using 'p:dataScroller' is a good idea, altought we need to rely on 'p:dataTable' since we need the sort and filter functions (and don't want to implement them on each site). Imho this should be fixed by the primeFaces team, since this really makes the component like unusable for some cases.

**Comment by malex666** [comment]
thanks for your answer. Good to know that it's not just us who do have problems with live scroll. Your approach using 'p:dataScroller' is a good idea, altought we need to rely on 'p:dataTable' since we need the sort and filter functions (and don't want to implement them on each site). Imho this should be fixed by the primeFaces team, since this really makes the component like unusable for some cases.

**Comment by melloware** [comment]
Fixed for 9.0 with this fix: https://github.com/primefaces/primefaces/issues/5230
